### PR TITLE
#103 Create User entity with service and repo 

### LINKED ISSUE #103: Users entity
Create users entity for accessing personalized list of movies

Name 
Surname
Birthdate

### PR DESCRIPTION
**Pull request by daweedek2**
Prepare user entity, service and repo for future creation of users by admin.
Closes #103

### CODE REVIEW

File: src/main/resources/data.sql [hunk 1]
```diff
@@ -2,9 +2,9 @@ INSERT INTO role (role_id, role)
 VALUES (101, "ADMIN"), (102, "USER")
 ON DUPLICATE KEY UPDATE role_id = role_id;
 
-INSERT INTO user_account (user_id, username, password)
-VALUES  (201, "admin", "$2y$12$YlXEdLOhX35AYO60kcC3Su8zJI0zKO/7vmS9QXrBPBhJRcx1uJZTi"),
-(202, "user", "$2y$12$YiMOV0G16lJGl4y.TPYi9.kQK9Pop1q4dnEppxW0Sg9iAk.mPcN46")
+INSERT INTO user_account (user_id, username, password, birthdate)
```
Comment (mjanousek) [inline src/main/resources/data.sql:5]: Aha, tak pokud tady je birthdate, tak je v tom formulari ve druhem requestu spatne jen ten label.

File: src/main/java/kostka/moviecatalog/configuration/DateConverter.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package kostka.moviecatalog.configuration;
+
+import javax.persistence.AttributeConverter;
+import javax.persistence.Converter;
+import java.time.LocalDate;
+import java.sql.Date;
+
+@Converter(autoApply = true)
+public class DateConverter implements AttributeConverter<LocalDate, Date> {
```
Comment (mjanousek) [inline src/main/java/kostka/moviecatalog/configuration/DateConverter.java:9]: Is is possible to use LocalDate on both sides? https://jdbc.postgresql.org/documentation/head/8-date-time.html

File: src/main/java/kostka/moviecatalog/dto/UserDto.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
+package kostka.moviecatalog.dto;
+
+import javax.validation.constraints.NotEmpty;
+
+public class UserDto {
```
Comment (crlikcngroup) [inline src/main/java/kostka/moviecatalog/dto/UserDto.java:5]: Just a thought: Such DTO classes might be final as well.

File: src/main/java/kostka/moviecatalog/service/UserService.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+import static kostka.moviecatalog.configuration.WebSecurityConfiguration.USER_ROLE;
+
+@Service
+public class UserService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(UserService.class);
+    private UserRepository userRepository;
+    private RoleRepository roleRepository;
+
+    BCryptPasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
```
Comment (crlikcngroup) [inline src/main/java/kostka/moviecatalog/service/UserService.java:28]: Was not this specified as a bean somewhere? I'd better use that.

Comment (daweedek2) [reply src/main/java/kostka/moviecatalog/service/UserService.java:28]: true, thanks

File: src/main/java/kostka/moviecatalog/configuration/DateConverter.java (hunk 2 above)
Comment (daweedek2) [reply src/main/java/kostka/moviecatalog/configuration/DateConverter.java:9]: in mysql database it is not possible afaik

File: src/main/java/kostka/moviecatalog/dto/UserDto.java (hunk 3 above)
Comment (daweedek2) [reply src/main/java/kostka/moviecatalog/dto/UserDto.java:5]: I can do it, yes

Comment (crlikcngroup) [reply src/main/java/kostka/moviecatalog/dto/UserDto.java]: But optional. Maybe a matter of taste.

**Pull request merged**